### PR TITLE
Migrate rest of forms to Final Form

### PR DESCRIPTION
**Pull request by lyyder**
Migrate remaining forms from Redux Form to Final Form and remove the `redux-form` dependency. Also now all the form components can be found in the src/forms folder.

The gzipped bundle size was reduced from __501.46 KB__ to __474.07 KB__.

### CODE REVIEW

File: src/components/SelectMultipleFilterPlain/SelectMultipleFilterPlainForm.js [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
+    <FinalForm
+      {...rest}
+      mutators={{ ...arrayMutators }}
+      onSubmit={vals => console.log('SelectMultipleFilterPlainForm', vals)}
+      render={({ className, name, options, twoColumns, onChange }) =>
+        <Form className={className}>
+        <FormSpy onChange={handleChange} subscription={{ values: true, dirty: true }} />
+          <FieldCheckboxGroup
+            name={name}
+            id={`${name}-filter`}
```
Comment (Gnito) [inline src/components/SelectMultipleFilterPlain/SelectMultipleFilterPlainForm.js]: I have used either same string as in the **name** props or if the form is used several times, I have passed in a **formId**

```
id={formId ? `${formId}.phoneNumber` : 'phoneNumber'}
```

File: src/components/SelectMultipleFilterPlain/SelectMultipleFilterPlainForm.js [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
-        name={name}
-        id={`${form}.${name}`}
-        options={options}
-        twoColumns={twoColumns}
-      />
-    </Form>
+    <FinalForm
+      {...rest}
+      mutators={{ ...arrayMutators }}
+      onSubmit={vals => console.log('SelectMultipleFilterPlainForm', vals)}
```
Comment (Gnito) [inline src/components/SelectMultipleFilterPlain/SelectMultipleFilterPlainForm.js]: This is probably unnecessary

**Comment by Gnito** [comment]
I didn't check this yet fully but looks OKish (you need to update changelog.md)
Also, it might be good to push this branch to stage and test with mobile phones.

File: src/forms/LocationSearchForm/LocationSearchForm.js [hunk 3]
```diff
@@ -1,55 +1,72 @@
...
-      onSubmit({ location });
+      props.onSubmit({ location });
     }
   };
 
-  const classes = classNames(rootClassName || css.root, className);
+  return (
+    <FinalForm
+      {...props}
+      render={({ rootClassName, className, intl, handleSubmit }) => {
```
Comment (Gnito) [inline src/forms/LocationSearchForm/LocationSearchForm.js]: We haven't used this spread format of props with other components. It helps to debug code if there's full prop list available.

File: src/forms/LocationSearchForm/LocationSearchForm.js [hunk 4]
```diff
@@ -1,55 +1,72 @@
...
+        const preventFormSubmit = e => e.preventDefault();
 
-  // Allow form submit only when the place has changed
-  const preventFormSubmit = e => e.preventDefault();
+        return (
+          <Form className={classes} onSubmit={preventFormSubmit}>
+            <Field
+              name="location"
+              format={null}
+              render={({ input, meta }) => {
```
Comment (Gnito) [inline src/forms/LocationSearchForm/LocationSearchForm.js]: We haven't used this spread format of props with other components. It helps to debug code if there's full prop list available.

File: src/forms/SelectMultipleFilterPlainForm/SelectMultipleFilterPlainForm.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      onChange(formState.values);
+    }
+  };
+
+  return (
+    <FinalForm
+      {...rest}
+      onSubmit={() => null}
+      mutators={{ ...arrayMutators }}
+      render={({ className, name, options, twoColumns, onChange }) => (
```
Comment (Gnito) [inline src/forms/SelectMultipleFilterPlainForm/SelectMultipleFilterPlainForm.js]: We haven't used this spread format of props with other components. It helps to debug code if there's full prop list available.

File: src/forms/TopbarSearchForm/TopbarSearchForm.js [hunk 6]
```diff
@@ -30,33 +29,57 @@ class TopbarSearchFormComponent extends Component {
...
+          const preventFormSubmit = e => e.preventDefault();
 
-    // Allow form submit only when the place has changed
-    const preventFormSubmit = e => e.preventDefault();
+          return (
+            <Form className={classes} onSubmit={preventFormSubmit}>
+              <Field
+                name="location"
+                format={null}
+                render={({ input, meta }) => {
```
Comment (Gnito) [inline src/forms/TopbarSearchForm/TopbarSearchForm.js]: We haven't used this spread format of props with other components. It helps to debug code if there's full prop list available.

**Comment by lyyder** [comment]
@Gnito The form props spreading/destructuring has now been changed so that all the forms provide the `formRenderProps` in the `render` function.

**Pull request merged**